v2/federation/handler.go

```diff
@@ -14,6 +16,27 @@ import (
 // expensive.
 // See the specification for more details:
 // https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE_Trust_Domain_and_Bundle.md
-func Handler(trustDomain spiffeid.TrustDomain, source spiffebundle.Source) http.Handler {
-	panic("not implemented")
+func Handler(trustDomain spiffeid.TrustDomain, source spiffebundle.Source, log logger.Logger) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			http.Error(w, "method is not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+
+		bundle, err := source.GetBundleForTrustDomain(trustDomain)
+		if err != nil {
+			log.Errorf("unable to get bundle for provided trust domain %q: %v", trustDomain, err)
+			http.Error(w, fmt.Sprintf("unable to get bundle for provided trust domain %q", trustDomain), http.StatusInternalServerError)
+			return
+		}
+		data, err := bundle.Marshal()
+		if err != nil {
+			log.Errorf("unable to marshal bundle for trust domain %q: %v", trustDomain, err)
+			http.Error(w, fmt.Sprintf("unable to marshal bundle for trust domain %q", trustDomain), http.StatusInternalServerError)
```

Comment:
I'd probably avoid leaking internal details over the HTTP interface. The logs can provide the necessary details to operators:
```suggestion
			http.Error(w, fmt.Sprintf("unable to serve bundle for %q", trustDomain), http.StatusInternalServerError)
```